cli/src/test.js

```diff
@@ -10,100 +7,119 @@ import {
   runDockerContainer,
 } from './execDocker/docker.js';
 import { readIDappConfig } from './utils/idappConfigFile.js';
-import { TEST_INPUT_DIR, TEST_OUTPUT_DIR } from './config/config.js';
-
-const execAsync = util.promisify(exec);
+import {
+  IEXEC_WORKER_HEAP_SIZE,
+  TEST_INPUT_DIR,
+  TEST_OUTPUT_DIR,
+} from './config/config.js';
+import { handleCliError } from './utils/cli-helpers.js';
 
 export async function test(argv) {
-  await cleanTestOutput();
-  if (argv.docker) {
-    await testWithDocker(argv.params);
-  } else {
-    await testWithoutDocker(argv.params);
+  const spinner = ora();
+  try {
+    await cleanTestOutput({ spinner });
+    await testApp({ args: argv.params, spinner });
+    // TODO check output files
+    // - output required files
+    // - outpout dir size
+    await askShowTestOutput({ spinner });
+  } catch (error) {
+    handleCliError({ spinner, error });
+  }
+}
+
+async function askShowTestOutput({ spinner }) {
+  // Prompt user to view result
+  const continueAnswer = await inquirer.prompt({
+    type: 'confirm',
+    name: 'continue',
+    message: `Would you like to see the result? (View ./${TEST_OUTPUT_DIR}/)`,
+  });
+  if (continueAnswer.continue) {
+    const files = await readdir(TEST_OUTPUT_DIR).catch(() => []);
+    if (files.length === 0) {
+      spinner.warn('output directory is empty');
+    } else {
+      spinner.info(
+        `output directory content:\n${files.map((file) => '  - ' + file).join('\n')}`
+      );
+    }
   }
 }
 
-async function cleanTestOutput() {
+async function cleanTestOutput({ spinner }) {
+  // just start the spinner, no need to persist success in terminal
+  spinner.start('Cleaning output directory...');
   await rm(TEST_OUTPUT_DIR, { recursive: true, force: true });
   await mkdir(TEST_OUTPUT_DIR);
 }
 
-async function testWithoutDocker(arg) {
-  const spinner = ora('Reading iDapp JSON config file ...').start();
+export async function testApp({ args = undefined, spinner }) {
+  const idappConfig = await readIDappConfig();
+  const { withProtectedData } = idappConfig;
 
-  const { withProtectedData } = await readIDappConfig(spinner);
-  spinner.succeed('Reading idapp JSON config file.');
+  // just start the spinner, no need to persist success in terminal
+  spinner.start('Checking docker daemon is running...');
+  await checkDockerDaemon();
+  // build a temp image for test
+  spinner.start('Building app docker image for test...\n');
+  const imageId = await dockerBuild({
+    isForTest: true, // Adjust based on your logic
```

Comment:
We can probably clean this comment

```suggestion
    isForTest: true,
```